op-node/rollup/derive/deposit_log.go

```diff
@@ -12,34 +12,55 @@ import (
 )
 
 var (
-	DepositEventABI     = "TransactionDeposited(address,address,uint256,uint256,uint64,bool,bytes)"
-	DepositEventABIHash = crypto.Keccak256Hash([]byte(DepositEventABI))
+	DepositEventABI      = "TransactionDeposited(address,address,uint256,bytes)"
+	DepositEventABIHash  = crypto.Keccak256Hash([]byte(DepositEventABI))
+	DepositEventVersion0 = common.Hash{}
 )
 
 // UnmarshalDepositLogEvent decodes an EVM log entry emitted by the deposit contract into typed deposit data.
 //
 // parse log data for:
 //     event TransactionDeposited(
-//    	 address indexed from,
-//    	 address indexed to,
-//       uint256 mint,
-//    	 uint256 value,
-//    	 uint64 gasLimit,
-//    	 bool isCreation,
-//    	 data data
+//         address indexed from,
+//         address indexed to,
+//         uint256 indexed version,
+//         bytes opaqueData
 //     );
 //
 // Additionally, the event log-index and
 func UnmarshalDepositLogEvent(ev *types.Log) (*types.DepositTx, error) {
-	if len(ev.Topics) != 3 {
-		return nil, fmt.Errorf("expected 3 event topics (event identity, indexed from, indexed to)")
+	if len(ev.Topics) != 4 {
+		return nil, fmt.Errorf("expected 4 event topics (event identity, indexed from, indexed to, indexed version), got %d", len(ev.Topics))
 	}
 	if ev.Topics[0] != DepositEventABIHash {
 		return nil, fmt.Errorf("invalid deposit event selector: %s, expected %s", ev.Topics[0], DepositEventABIHash)
 	}
-	if len(ev.Data) < 6*32 {
-		return nil, fmt.Errorf("deposit event data too small (%d bytes): %x", len(ev.Data), ev.Data)
+	if len(ev.Data) < 64 {
+		return nil, fmt.Errorf("incomplate opaqueData slice header (%d bytes): %x", len(ev.Data), ev.Data)
 	}
+	if len(ev.Data)%32 != 0 {
+		return nil, fmt.Errorf("expected log data to be multiple of 32 bytes: got %d bytes", len(ev.Data))
+	}
+
+	// indexed 0
+	from := common.BytesToAddress(ev.Topics[1][12:])
+	// indexed 1
+	to := common.BytesToAddress(ev.Topics[2][12:])
+	// indexed 2
+	version := ev.Topics[3]
+	// unindexed data
+	var opaqueContentOffset uint256.Int
+	opaqueContentOffset.SetBytes(ev.Data[0:32])
+	if !opaqueContentOffset.IsUint64() || opaqueContentOffset.Uint64() != 32 {
```

Comment:
Can use a comment to explain the fixed offset `32`. Is the offset from the start of `ev.Data`, and if so shouldn't it be `64` considering we're reading `ev.Data[64+...` down below?

Reply:
Considering the var is called `opaqueContentOffset`, it seems that the offset must indicate start of the content.

Reply:
I see that the offset `0x20` is document below. Going to give the benefit of doubt and assume it means content including the length.